OpenRA.Mods.Common/Traits/World/EditorActorPreview.cs

```diff
@@ -91,19 +81,36 @@ public EditorActorPreview(WorldRenderer worldRenderer, string id, ActorReference
 			terrainRadarColorInfo = Info.TraitInfoOrDefault<RadarColorFromTerrainInfo>();
 			UpdateRadarColor();
 
-			// Bounds are fixed from the initial render.
-			// If this is a problem, then we may need to fetch the area from somewhere else
+			GenerateBounds();
+		}
+
+		void GenerateBounds()
+		{
 			var r = previews.SelectMany(p => p.ScreenBounds(worldRenderer, CenterPosition));
 
 			Bounds = r.Union();
 
-			SelectionBox = new SelectionBoxAnnotationRenderable(new WPos(CenterPosition.X, CenterPosition.Y, 8192),
+			selectionBox = new SelectionBoxAnnotationRenderable(new WPos(CenterPosition.X, CenterPosition.Y, 8192),
 				new Rectangle(Bounds.X, Bounds.Y, Bounds.Width, Bounds.Height), Color.White);
 
 			// TODO: updating all actors on the map is not very efficient.
 			onCellEntryChanged = _ => GeneratePreviews();
```

Comment:
this should also trigger a visual update. When map height changes so should the actors position. Easiest way to test is placing cliffs and having terrain overlay turned on

Reply:
Oh yes, I see - also pasting terrain of a different height with the actor filter turned off